### PR TITLE
Slides: empty-placeholder 1-click text-edit entry (P1.4)

### PR DESCRIPTION
**Pull request by hackerwins**
## Summary

Phase B (P1.4) of the [hover and text-edit entry parity spec](docs/design/slides/slides-hover-and-text-edit-entry.md). A fresh non-shift click on an empty layout placeholder now both selects AND enters text-edit in the same gesture — a brand-new "Title + Body" slide is typeable in one click per region instead of click-then-double-click.

- Pure `isEmptyPlaceholder(element)` predicate in `packages/slides/src/view/editor/interactions/select.ts`. Delegates to the existing `isElementEmpty` so the predicate stays in lockstep with the renderer's ghost-hint gate (single source of truth — if the user sees a ghost, the predicate returns `true`).
- Wired into `SlidesEditor.onPointerDown` fresh-selection branch — routes to `enterEditMode(...)` instead of `startDrag(...)` when the predicate fires.
- Scoped to text elements with `placeholderRef` and a functionally-empty body. User-authored empty text boxes keep select-only behavior (deliberate divergence from Google Slides per spec § Risks).

## Deviations from spec letter

- Spec said "zero blocks, or a single empty paragraph block." The implementation matches the renderer's broader `isBlocksEmpty` gate (zero blocks, or every inline empty across any block kind/count). This is intentional — it keeps the predicate and the rendered ghost hint impossible to drift apart.
- Spec said `selection.replace([id]) then enterEditMode() in the same pointer-up handler`. Implementation runs both in `onPointerDown`, where the existing fresh-selection branch already lives. If dogfooding shows users want to drag empty placeholders without entering edit, deferring to pointer-up is a small follow-up (Phase C will touch the no-drag pointer-up classifier anyway).

Both deviations are captured in `docs/tasks/active/20260602-slides-hover-text-edit-phase-b-lessons.md`.

## Out of scope (follow-up PRs per spec)

- Phase C (P1.5 slow double-click), Phase D (P2.6 first-char forwarding), Phase E (P2.7 edge-zone resize cursor) — see umbrella todo.
- Browser harness scenario — still deferred for the slides interaction harness (same blocker as Phase A's A6 in #331).

## Test plan

- [x] `pnpm verify:fast` green (877 tests; 0 exit)
- [x] Predicate truth-table in `select.test.ts` — 8 cases (presence/absence of `placeholderRef`, zero/one/multi blocks, non-text, null/undefined)
- [x] Integration in new `empty-placeholder-entry.test.ts` — 5 cases (positive entry, non-empty placeholder, non-placeholder textbox, shift-click suppression, already-selected short-circuit)
- [x] Code-review skill — Important findings addressed inline (predicate doc-comment alignment, plan Risks-table accuracy, test helper readability, multi-block coverage)
- [x] Manual smoke in `pnpm dev` — deferred to reviewer; see Task B4 Step 3 of `docs/tasks/active/20260602-slides-hover-text-edit-phase-b-todo.md` for the 7-step smoke script. Step 7 specifically probes the pointer-down-vs-pointer-up question above.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Empty layout placeholders in the slides editor now support one‑click editing: clicking an empty placeholder selects it and immediately enters text‑edit mode; non‑empty placeholders and regular text boxes remain select‑only.

* **Bug Fixes**
  * Click handling now prevents default focus‑stealing on initial pointer down for empty placeholders.

* **Tests**
  * Added unit and integration tests covering empty‑placeholder detection, one‑click entry, shift‑click behavior, and selection edge cases.

* **Documentation**
  * Updated interaction spec and added planning/lessons docs detailing the behavior and verification steps.

### CODE REVIEW

File: packages/slides/test/view/editor/empty-placeholder-entry.test.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  const el = slide.elements.find((e) => e.placeholderRef?.type === type)!;
+  return el.id;
+}
+
+describe('empty-placeholder 1-click entry', () => {
+  let editor: SlidesEditor | null = null;
+
+  beforeEach(() => {
+    if (editor) { editor.detach(); editor = null; }
+  });
```
[the remark this answers is not in the document]
Comment (hackerwins) [reply packages/slides/test/view/editor/empty-placeholder-entry.test.ts]: Applied in 752c91e7 — switched to `afterEach`. The final test's editor now detaches like the rest, so the document-level pointerup/pointercancel capture listeners can't leak across files.

File: docs/tasks/active/20260602-slides-hover-text-edit-phase-b-todo.md [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+- Modify: `docs/design/slides/slides.md` (Interactions table)
+- Modify: `docs/tasks/active/20260601-slides-hover-text-edit-entry-todo.md` (replace Phase B stub with pointer)
+
+- [ ] **Step 1: Append row to `slides.md` Interactions table**
+
+Locate the Interactions table (search for "Cursor over selected text region" — added in Phase A — and append after it):
+
+```
+| Click empty layout placeholder | first click on a `text` element with `placeholderRef` and empty body | selects AND `enterEditMode(...)`; see [slides-hover-and-text-edit-entry.md § P1.4](slides-hover-and-text-edit-entry.md) |
+```
```
[the remark this answers is not in the document]
Comment (hackerwins) [reply docs/tasks/active/20260602-slides-hover-text-edit-phase-b-todo.md:538]: Skipping: this repo has no `.markdownlint*` config and the sibling Phase A todo (`20260601-slides-hover-text-edit-entry-todo.md`) has 38 bare `````` fences in active use. MD040 isn't a project convention; the bare prose fence matches established practice here. Re-visit if a markdownlint step is added to CI.

**Comment by hackerwins** [comment]
CodeRabbit review processed in 752c91e7:

- **afterEach teardown** (Major) — applied; the final test's editor now detaches cleanly.
- **Cross-region regression test** (nitpick from the review summary) — added an explicit `enters edit when clicking an empty placeholder while a different element is selected` case. This is the title→body flow Phase B was built for, so worth a dedicated assertion.
- **MD040 fence languages** (Minor) — skipped. Repo has no `.markdownlint*` config; sibling Phase A todo uses 38 bare `````` fences. Re-visit if a markdownlint step is added to CI.

File: packages/slides/test/view/editor/empty-placeholder-entry.test.ts [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+    editor.setSelection([titleId]);
+
+    click(
+      canvas,
+      body.frame.x + body.frame.w / 2,
+      body.frame.y + body.frame.h / 2,
+    );
+
+    expect(editor.getEditingElementId()).toBe(bodyId);
+    expect(editor.getSelection()).toEqual([bodyId]);
```
[the remark this answers is not in the document]
Comment (hackerwins) [reply packages/slides/test/view/editor/empty-placeholder-entry.test.ts]: Fair point — applied. Test now drives the realistic flow: click empty Title → `editor.exitTextEditing()` to commit → click empty Body. Asserts each step's `editingElementId` + `getSelection()`. Would have caught any regression where lingering edit state blocks the next 1-click entry.
